internal/cmd/use_test.go

```diff
@@ -73,3 +77,172 @@ func TestFuncEUse_InstallsAndWritesHomeVersion(t *testing.T) {
 	require.NoError(t, err)
 	require.Equal(t, o.EnvoyVersion, version.Version(f))
 }
+
+func TestFuncEUse_InstallMinorVersion(t *testing.T) {
+	o, cleanup := setupTest(t)
+	defer cleanup()
+
+	type testCase struct {
+		name           string
+		firstVersions  avaliableVersions
+		secondVersions avaliableVersions
+		minorVersion   string
+	}
+
+	tests := []testCase{
+		{
+			name: "upgradable",
+			firstVersions: avaliableVersions{
+				latestPatch: "3",
+				versions:    []version.Version{"1.18.3"},
+			},
+			secondVersions: avaliableVersions{
+				latestPatch: "4",
+				versions:    []version.Version{"1.18.3", "1.18.4"},
+			},
+			minorVersion: "1.18",
+		},
+		{
+			name: "not-upgraded",
+			firstVersions: avaliableVersions{
+				latestPatch: "3",
+				versions:    []version.Version{"1.12.3"},
+			},
+			secondVersions: avaliableVersions{
+				latestPatch: "3",
+				versions:    []version.Version{"1.12.3"},
+			},
+			minorVersion: "1.12",
+		},
+	}
+
+	for _, tc := range tests {
+		tc := tc
+		t.Run(tc.name, func(t *testing.T) {
+			var err error
+			o.FuncEVersions, err = newFuncEVersionsTester(o, tc.firstVersions)
+			require.NoError(t, err)
+
+			c, _, _ := newApp(o)
+			require.NoError(t, c.Run([]string{"func-e", "use", tc.minorVersion}))
+			f, err := os.ReadFile(filepath.Join(o.HomeDir, "version"))
+			require.NoError(t, err)
+			require.Equal(t, version.Version(tc.minorVersion), version.Version(f))
+
+			// Set o.EnvoyVersion to empty string so the logic for ensuring installed Envoy version works.
+			o.EnvoyVersion = ""
+			c, stdout, stderr := newApp(o)
+			require.NoError(t, c.Run([]string{"func-e", "which"}))
+			envoyPath := filepath.Join(o.HomeDir, "versions", tc.minorVersion+"."+tc.firstVersions.latestPatch, "bin", "envoy"+moreos.Exe)
+			require.Equal(t, moreos.Sprintf("%s\n", envoyPath), stdout.String())
+			require.Empty(t, stderr)
+
+			// Update the map returned by Get.
+			o.FuncEVersions, err = newFuncEVersionsTester(o, tc.secondVersions)
+			require.NoError(t, err)
+			c, _, _ = newApp(o)
+			require.NoError(t, c.Run([]string{"func-e", "use", tc.minorVersion}))
+			f, err = os.ReadFile(filepath.Join(o.HomeDir, "version"))
+			require.NoError(t, err)
+			require.Equal(t, version.Version(tc.minorVersion), version.Version(f))
+
+			// Set o.EnvoyVersion to empty string so the logic for ensuring installed Envoy version works.
+			o.EnvoyVersion = ""
+			c, stdout, stderr = newApp(o)
+			require.NoError(t, c.Run([]string{"func-e", "which"}))
+			envoyPath = filepath.Join(o.HomeDir, "versions", tc.minorVersion+"."+tc.secondVersions.latestPatch, "bin", "envoy"+moreos.Exe)
+			require.Equal(t, moreos.Sprintf("%s\n", envoyPath), stdout.String())
+			require.Empty(t, stderr)
```

Comment:
Please let me know if you want me to wrap some of these. I don't know the preference for wrapping codes for tests. Some projects asked me to wrap, but others want it easy to read, hence copy-pasting is fine.

Reply:
I think this looks fine as-is! Thank you!

Reply:
I think this looks fine as-is! Thank you!